Modules/Microsoft365DSC/DSCResources/MSFT_AADUser/MSFT_AADUser.psm1

```diff
@@ -462,51 +462,58 @@ function Set-TargetResource
         }
         #endregion
 
-        if ($null -ne $Password)
+        if ($user.UserPrincipalName)
         {
-            $passwordValue = $Password.GetNetworkCredential().Password
+            Write-Verbose -Message "Updating Office 365 User $UserPrincipalName Information"
+
+            if ($null -ne $Password)
+            {
+                Write-Verbose -Message "PasswordProfile property will not be updated"
+            }
+
+            $CreationParams.Add('UserId', $UserPrincipalName)
+            Update-MgUser @CreationParams
         }
         else
         {
-            try
+
+            if ($null -ne $Password)
             {
-                # This only works in PowerShell 5.1
-                $passwordValue = [System.Web.Security.Membership]::GeneratePassword(30, 2)
+                $passwordValue = $Password.GetNetworkCredential().Password
             }
-            catch
+            else
             {
-                $TokenSet = @{
-                    U = [Char[]]'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
-                    L = [Char[]]'abcdefghijklmnopqrstuvwxyz'
-                    N = [Char[]]'0123456789'
-                    S = [Char[]]'!"#$%&''()*+,-./:;<=>?@[\]^_`{|}~'
+                try
+                {
+                    # This only works in PowerShell 5.1
+                    $passwordValue = [System.Web.Security.Membership]::GeneratePassword(30, 2)
```

Comment:
This will only work if you first run `Add-Type -AssemblyName System.Web`

Reply:
assembly will now be loaded

Reply:
Any chance to add a check, if we are in PS5?

Reply:
we can check for $PSVersionTable.PSVersion.Major -eq 5

Reply:
That is a good idea. Can you change this implementation?